### PR TITLE
add note regarding declarative schema

### PR DESCRIPTION
**Pull request by dfelton**
## Purpose of this pull request

Add note and hyperlink regarding declarative schema. Discourage usage of upgrade scripts in the extension lifecycle.

## Affected pages

-  https://developer.adobe.com/commerce/php/development/prepare/extension-lifecycle/#disabled-modules

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My changes follow the code style of this project.
- [x] I have read the **CONTRIBUTING** document.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/pages/development/prepare/extension-lifecycle.md [hunk 1]
```diff
@@ -5,6 +5,8 @@ description: Learn how to create classes that execute code when Adobe Commerce a
 
 # Extension lifecycle
 
+*Note:* The recommendations laid out below are to be phased out in the future in favor of [declarative schema](https://developer.adobe.com/commerce/php/development/components/declarative-schema/#declarative-schema) development approaches.
```
Comment (jeff-matthews) [inline src/pages/development/prepare/extension-lifecycle.md:8]:
```suggestion
<InlineAlert variant="info" slots="text"/>

The recommendations laid out below are to be phased out in the future in favor of [declarative schema](https://developer.adobe.com/commerce/php/development/components/declarative-schema/#declarative-schema) development approaches.
```

**Review by jeff-matthews — CHANGES_REQUESTED** [review]
Thanks for the contribution @dfelton! I made a suggestion to use the [inline alert block](https://github.com/adobe/aio-theme#inline-alert-block) component.

**Comment by dfelton** [comment]
ok done dinking around heh... my OCD desired to change my email on the commit as well as gpg sign it.

**Review by jeff-matthews — APPROVED** [review]
Nice! Thanks again @dfelton!

**Pull request merged**